lib/seahorn/BvOpSem2RawMemMgr.cc

```diff
@@ -642,6 +642,17 @@ RawMemManagerCore::MemValTy RawMemManagerCore::MemSet(PtrTy ptr, Expr _val,
 RawMemManagerCore::MemValTy RawMemManagerCore::MemSet(PtrTy ptr, Expr _val,
                                                       Expr len, MemValTy mem,
                                                       uint32_t align) {
+  auto simplifiedLength = len;
+  if (m_ctx.shouldSimplify()) {
```

Comment:
not sure the simplification should depend on this flag.

This flag is for aggressive simplification of memory state during vcgen. I would imagine that simplification in this step should be done aggressively at all time.

If this is a problem, might restrict simplifications only to cases where `len` has small dag size. 

However, I do believe that simplifying all the time should be doable.

Reply:
I see what you mean. However, currently `shouldSimplify` is also used elsewhere. For example in simplifying isDeref: https://sourcegraph.com/github.com/seahorn/seahorn@dev14/-/blob/lib/seahorn/BvOpSem2ExtraWideMemMgr.cc?L122

Having a single flag to turn simplification on/off is a useful tool for debugging. If we want to keep a memory-state only simplification flag then should be break it into two flags -- `shouldSimplify` and `shouldSimplifyMemory` and only simplify memory if both are on.

Reply:
Yes, that's another incorrect use of that option. The external documentation for the option is what you call `shouldSimplifyMemory`. 

So perhaps you need to add another flag. 

We find that in some Rust verification aggressive simplification is very very expensive. So we turn it off. However, we still would like is_deref and memcpy checks to be simplified!

Reply:
Added a new flag `shouldSimplifyNonMem`.